### PR TITLE
Support 10X and Smartseq indexes

### PR DESCRIPTION
**Pull request by chuan-wang**

### CODE REVIEW

File: LIMS2DB/classes.py [hunk 1]
```diff
@@ -211,8 +211,13 @@ def __init__(self, session, log, step):
     def extract_barcode(self, chain):
         barcode=''
         bcp = re.compile("[ATCG\-]{4,}")
+        TENX_PAT = re.compile("SI-(?:GA|NA)-[A-H][1-9][0-2]?")
+        ST_PAT = re.compile("SI-TT-[A-H][1-9][0-2]?")
+        SMARTSEQ_PAT = re.compile('SMARTSEQ[1-9]?-[1-9][0-9]?[A-P]')
         if "NoIndex" in chain:
             return chain
+        if TENX_PAT.match(chain) or TENX_PAT.match(chain) or SMARTSEQ_PAT.match(chain):
```
Comment (alneberg) [inline LIMS2DB/classes.py:219]: I think there's a typo here, where TENX_PAT is used twice and ST_PAT is missing?

File: LIMS2DB/classes.py [hunk 2]
```diff
@@ -1008,8 +1013,13 @@ def get_library_preps(self, sample):
     def extract_barcode(self, chain):
         barcode=''
         bcp = re.compile("[ATCG\-\_]{4,}")
+        TENX_PAT = re.compile("SI-(?:GA|NA)-[A-H][1-9][0-2]?")
+        ST_PAT = re.compile("SI-TT-[A-H][1-9][0-2]?")
+        SMARTSEQ_PAT = re.compile('SMARTSEQ[1-9]?-[1-9][0-9]?[A-P]')
         if "NoIndex" in chain:
             return chain
+        if TENX_PAT.match(chain) or TENX_PAT.match(chain) or SMARTSEQ_PAT.match(chain):
```
Comment (alneberg) [inline LIMS2DB/classes.py:1021]: Same here

File: LIMS2DB/classes.py (hunk 1 above)
Comment (chuan-wang) [reply LIMS2DB/classes.py:219]: Ah, sorry!

**Review by aanil — APPROVED** [review]
👍 Looks ok to me

**Review by alneberg — APPROVED** [review]
Looks good!

**Pull request merged**